zebROS_ws/src/tf_object_detection/src/object_detection.cpp

```diff
@@ -26,30 +29,288 @@ void camera_info_callback(const sensor_msgs::CameraInfoConstPtr &info)
 	caminfovalid = true;
 }
 
-// Get the average of the values in the cv::Mat depth contained within
-// the supplied bounding rectangle
-double avgOfDepthMat(const cv::Mat& depth, const cv::Rect& bound_rect)
+// Find the median of a continuous cv::Mat
+float findMedianOfMat(cv::Mat mat) {
+	float median = 0;
+	if (mat.isContinuous()) {
+		// copy matrix data to a vector
+		std::vector<float> vec;
+		mat = mat.reshape(0, 1);
+		mat.copyTo(vec);
+		// remove 255 (when this is called, values that are masked are 255)
+		vec.erase(std::remove(vec.begin(), vec.end(), 255), vec.end());
+		// sort vector
+		std::sort(vec.begin(), vec.end()); // forgot this earlier
+		if ((vec.size() % 2) != 0) { // if odd
+			auto it = vec.begin() + vec.size()/2;
+			std::nth_element(vec.begin(), it, vec.end());
```

Comment:
Is nth_element redundant if the vector is already sorted?

Reply:
Good point -- I removed the `std::sort` and kept the `nth_element`. Thanks!